frontend/src/metabase/hooks/use-confirm-route-leave-modal.ts

```diff
@@ -64,11 +63,37 @@ export const useConfirmRouteLeaveModal = ({
     return removeLeaveHook;
   }, [isLocationAllowed, router, route, isEnabled, isConfirmed]);
 
-  useEffect(() => {
-    if (isConfirmed && nextLocation) {
-      dispatch(push(nextLocation));
```

Comment:
We were dispatching wrong action here and it caused a weird bug that when you clicked "Discard changes" (see "How to verify" section in PR description) the page header would be rendered (as if it was viz) but you'd still be in model metadata editing mode.

The repro accounts for that.